nixos/modules/virtualisation/google-compute-config.nix

```diff
@@ -69,6 +69,69 @@ in
   # GC has 1460 MTU
   networking.interfaces.eth0.mtu = 1460;
 
+  systemd.services.fetch-ssh-keys = {
+    description = "Fetch host keys and authorized_keys for root user";
+
+    wantedBy = [ "sshd.service" ];
+    before = [ "sshd.service" ];
+    after = [ "network-online.target" ];
+    wants = [ "network-online.target" ];
+
+    script =
+      let
+        wget = "${pkgs.wget}/bin/wget --retry-connrefused -t 15 --waitretry=10 --header='Metadata-Flavor: Google'";
+        mktemp = "mktemp --tmpdir=/run";
+      in ''
+        # When dealing with cryptographic keys, we want to keep things private.
+        umask 077
+        mkdir -m 0700 -p /root/.ssh
+
+        echo "Obtaining SSH keys..."
+        AUTH_KEYS=$(${mktemp})
+        ${wget} -O $AUTH_KEYS http://metadata.google.internal/computeMetadata/v1/instance/attributes/sshKeys
```

Comment:
Just to confirm, this is not shown in `http://metadata.google.internal/computeMetadata/v1/oslogin/users?username=root`, but a separate list?

Reply:
No, this metadata is added by the gce nixops plugin, which doesn't yet support oslogin.